packages/account/src/providers/transaction-request/transaction-request.ts

```diff
@@ -709,4 +711,39 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
   byteLength(): number {
     return this.toTransactionBytes().byteLength;
   }
+
+  /**
+   * WARNING: Do not enable without fully understanding the implications.
+   *
+   * Enables asset burn for the transaction request.
+   *
+   * @param burnEnabled - Whether the transaction request should enable asset burn.
+   * @returns This transaction request.
+   */
+  enableBurn(burnEnabled: boolean = true): this {
+    this.burnEnabled = burnEnabled;
+    return this;
+  }
+
+  /**
+   * @hidden
+   *
+   * Checks if the transaction request has burnable assets.
+   *
+   * @returns a boolean indicating whether the transaction request has burnable assets.
+   */
+  hasBurnableAssets(): boolean {
+    if (this.burnEnabled) {
```

Comment:
The way this method is named, IMO it shouldn't be concerned with the `burnEnabled` flag as it's just a query to see if the tx request has burnable assets. I'd say remove this `if(this.burnEnabled)` check and update [the provider](https://github.com/FuelLabs/fuels-ts/pull/3540/files#diff-69a6670d0426c74935ed2830ce552e7df2379c2d112a582306e57569fcd17029R870) to check:
```ts
if(!tx.burnEnabled && tx.hasBurnableAssets) { throw new FuelError(...) }
```